pintracker/stateless/stateless.go

```diff
@@ -299,59 +320,65 @@ func (spt *Tracker) Status(ctx context.Context, c cid.Cid) *api.PinInfo {
 		return oppi
 	}
 
+	pinInfo := &api.PinInfo{
+		Cid:      c,
+		Peer:     spt.peerID,
+		PeerName: spt.peerName,
+		TS:       time.Now(),
+	}
+
 	// check global state to see if cluster should even be caring about
 	// the provided cid
-	var gpin api.Pin
-	err := spt.rpcClient.Call(
-		"",
-		"Cluster",
-		"PinGet",
-		c,
-		&gpin,
-	)
-	if err != nil {
-		if rpc.IsRPCError(err) {
-			logger.Error(err)
-			return &api.PinInfo{
-				Cid:    c,
-				Peer:   spt.peerID,
-				Status: api.TrackerStatusClusterError,
-				Error:  err.Error(),
-				TS:     time.Now(),
+	gpin := &api.Pin{}
+	var err error
+	if noState(spt.state) {
+		err = spt.rpcClient.CallContext(
+			ctx,
+			"",
+			"Cluster",
+			"PinGet",
+			c,
+			gpin,
+		)
+		if err != nil {
+			if rpc.IsRPCError(err) {
+				logger.Error(err)
+				pinInfo.Status = api.TrackerStatusClusterError
+				pinInfo.Error = err.Error()
+				return pinInfo
 			}
+			// not part of global state. we should not care about
+			pinInfo.Status = api.TrackerStatusUnpinned
+			return pinInfo
+		}
+	} else {
+		gpin, err = spt.state.Get(ctx, c)
+		if err == state.ErrNotFound {
+			pinInfo.Status = api.TrackerStatusUnpinned
+			return pinInfo
 		}
-		// not part of global state. we should not care about
-		return &api.PinInfo{
-			Cid:    c,
-			Peer:   spt.peerID,
-			Status: api.TrackerStatusUnpinned,
-			TS:     time.Now(),
+		if err != nil {
+			logger.Error(err)
+			return nil
```

Comment:
NEVER return nil from these methods. We will switch back to not using pointers here. Return a valid PinInfo with an error set.

Reply:
changed this. Any already existing `return nil` is kept as it is.